### DISCUSSION TITLE
64 bit build for desktop

### DISCUSSION THREAD

**Discussion by TomasaMuller** [airsdk/Adobe-Runtime-Support#953] [Q&A]
I have a windows desktop AIR app (Main swf) that loads another swf at runtime.
Other swf is with swf-version 20.

Main SWF I have re-built with swf-version 40 with AIR SDK 29

I would like to know, to build a 64 bit version of the AIR app is it mandatory to re-build all the swf's that are being loaded by the Main SWF to swf-version 40.

Or is it just fine if only the main SWF is rebuild (as I have done above).

**Comment by ajwfrost** [comment]
I don't think any of the SWFs need to be rebuilt; the 32-bit or 64-bit option is just based on the runtime that's used to run them.
So if you run with adl.exe, it will be running in a 32-bit process, if you run with adl64.exe, it will be running in a 64-bit process, the SWF files though would be exactly the same ("int" is always a 32-bit type in ActionScript).
When building a Windows bundle/native package, you can just add the "-arch x64" to make them use the 64-bit runtime.

The only time this should make a difference is if you're using AIR Native Extensions, which need to have an implementation that matches the runtime's architecture..

thanks

**Comment by TomasaMuller** [comment]
@ajwfrost 

Thanks for your explanation.

The Air SDK 29.0 creates a captive runtime alongside the packaged app executable, I guess you are referring to this captive runtime being "64 bit" capable.
I didn't find any 64 bit adl64, and I don't think Adobe released any 64 bit debugger in the Air SDK 29or even later versions, All i find is "adl.exe" which I think is the 32 bit version of the debugger.

I read on adobe forums some had outlined how to build a 64bit version of the app, which I'm pasting here, as quoted.

> 
> Im using FlashDevelop so I will add the notes accordingly.
> 
> So to recap then and to build a 64bit windows application using the latest SDK (Download Adobe AIR 29 Beta - Adobe Labs)
> 
> Step 1
> 
> Make sure you have downloaded the correct SDK and you have added it to your project
> 
> Step 2
> 
> Add "-swf-version=40" (without quotes) into the Project properties -> compiler options -> Addition compiler options
> 
> OR
> 
> Add this to your .as3proj file <option additional="-swf-version=40" /> (if you do it this way, you will have to close and then re-open your project for this value to take effect.
> 
> Step 3
> 
> Add the NEW <architecture>64</architecture> into your application.xml file. I put this just below the <name> node
> 
> If you don't add this new node into your application.xml file, it will default to a 32bit application
> 
> Step 4
> 
> Build your Air Application. I used a batch file to do this.

In that answer the user specifically highlights 2 issues,
1. swf-version=40 should be set
2. New tag <architecture>64</architecture> should be placed in application.xml.
In is these points that  got warning bells ringing in my head to rebuild the swf's.

Please let me know what are your views regarding these.
Thanks.

**Reply by ajwfrost** [reply]
Okay so yes, you're right that Adobe didn't release a 64-bit ADL version so if you're just launching via that, then you'll always run 32-bit.

But you're also correct that you can package up with the 64-bit version of the captive runtime, and have your generated .exe file also being 64-bit. It should mean that in the generated folder, you'll have an "Adobe AIR" folder that's got the same binaries in as the one from your SDK under the "runtimes\air-captive\win64" folder.

I don't actually know exactly what they're doing with the new '64' tag in the application XML file, maybe there's an additional batch file they're running that I don't have with my FlashDevelop version? I believe FlashDeveloper's "PackageApp" script would create a .air file; but in order to create the bundle with captive runtime, you could perhaps do
```
call adt -package -target bundle -arch x64 %SIGNING_OPTIONS% %OUTPUT% %APP_XML% %FILE_OR_DIR%
```
instead of
```
call adt -package %OPTIONS% %SIGNING_OPTIONS% %OUTPUT% %APP_XML% %FILE_OR_DIR%
```

But I still wouldn't have said you need to re-build the SWF file with any specific version number for this to work..!

thanks

**Reply by TomasaMuller** [reply]
Thanks for the confirmation.

I was facing some issues when running the 64 bit version of the app. Basically I see 2 to 3x disk writes and UI freeze happening as compared to 32 bit version. 

However since you rule out swf-version of the SWF's in this scenario, I would have to look elsewhere.

Are you aware of any issues regarding disk operations for Air 29+
Since Air 29 has no 64bit debugger I cannot debug the app in this mode and so have to rely on other ways to reach some conclusions.

Does Harman Air have 64 bit debugger in their SDK?

Thanks again.

**Reply by ajwfrost** [reply]
@TomasaMuller sorry for the delay: if there are issues in the 64-bit version it may just be bugs in that version of the runtime, but the SWF itself shouldn't make a difference. I don't think we had any Windows-based updates for file handling, although just checking through the release notes and #134 is mentioned in case that might be similar?

Yes we have a 64-bit Windows debugger version in the 33.1 SDK.

thanks

**Comment by Jason0247** [comment]
@ajwfrost Hello, how to configure if you use animate CC?
There are no other options when publishing. After publishing, it is a 32-bit program. The SDK used is 33.1.620

**Reply by ajwfrost** [reply]
I'm not sure that Animate has this option ... I'll raise it with them, and we can look at adding an override option into the app descriptor file...

**Reply by Jason0247** [reply]
If you can add and publish 64 bit win running package in the application descriptor, it will be perfect. I'm looking forward to it.
